packages/opencensus-core/test/test-root-span.ts

```diff
@@ -77,6 +77,28 @@ describe('RootSpan', () => {
     });
   });
 
+  describe('nested spans', () => {
+    it('should get nested spans from rootspan instance', () => {
+      const root = new RootSpan(tracer, name, kind, traceId, parentSpanId);
+      root.start();
+      assert.equal(root.numberOfChildren, 0);
+      const child =
+          root.startChildSpan('childName', types.SpanKind.UNSPECIFIED);
+      assert.equal(root.numberOfChildren, 1);
```

Comment:
My understanding is that `numberOfChildren` is supposed to measure the number of *direct* children, which enables checks for trace completeness (if you all spans have child spans for all their children then you know the whole trace has been ingested). But making a span a parent of something other than the root breaks that contract.

My [suggestion for how to fix that](https://github.com/census-instrumentation/opencensus-node/issues/105#issuecomment-468304948) would require some additional changes to make the `Span` and `RootSpan` classes be more similar.

But we might be able to do something simpler in the short run - what if rather than passing in a parent span ID, it passed in a parent span, which would then have its child count incremented?

Reply:
Ah, maybe that is why the Zipkin UI still shows the grandchildren as cascading rather than nesting.

Reply:
Possibly (though that would somewhat surprise me), is there a way you can look at the raw trace to make sure all the parent links seem correct?

Reply:
I found that the issue was that `shared` needs to be false for all spans other than the root span:
https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-exporter-zipkin/src/zipkin.ts#L169

I'll include this in my PR.

Reply:
@draffensperger, you suggested changing the child count of the appropriate parent. Currently, only `RootSpan` has `numberOfChildren`. Are you suggesting that `numberOfChildren` move to `SpanBase`?
Also, what should happen when a non-root span ends? Should all of its children be truncated like is done in `RootSpan.end()`? That would mean that `spansLocal` would also move to `SpanBase`.

Reply:
Yes, I believe moving `numberOfChildren` to `SpanBase` makes sense. I do think that moving the `spansLocal` to `SpanBase` also makes sense in addition to some other refactors I describe in https://github.com/census-instrumentation/opencensus-node/issues/105#issuecomment-468304948

Reply:
> I found that the issue was that shared needs to be false for all spans other than the root span.

I think `shared` needs to be `True` if span started by another tracer (on a different host). What's the plan for handling this scenerio, if we change to `False`. Also, it would be nice to have a seperate PR to handle this change.

Reply:
> Possibly (though that would somewhat surprise me), is there a way you can look at the raw trace to make sure all the parent links seem correct?

I think it is possible with [console](https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/exporters/console-exporter.ts#L63) exporter, need to add parentId in the result.

Reply:
@mayurkale22 The Zipkin UI has a button you can click that provides the JSON.  Super easy.

Reply:
@draffensperger I sketched out the consolidation of Span and RootSpan in #441.  Please let me know if that is on the right track.